libfprint-2-tod1-vfs0090: init at 0.8.5

Valodim opened the pull request:
###### Motivation for this change

add vfs0090 tod module, which allow usage of the fingerprint sensor built into 2016 thinkpad, such as the T460p

upstream: https://gitlab.freedesktop.org/3v1n0/libfprint-tod-vfs0090

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

Valodim committed:
libfprint-2-tod1-vfs0090: init at 0.8.5

r-rmcgibbo commented:
Result of `nixpkgs-review pr 126403` at a75c6664 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>libfprint-2-tod1-vfs0900</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/patch-phase.md">patch-phase</a>

`patchPhase` should not be overridden, use `postPatch` instead.

Near pkgs/development/libraries/libfprint-2-tod1-vfs0090/default.nix:16:3:
```
   |
16 |   patchPhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/development/libraries/libfprint-2-tod1-vfs0090/default.nix:27:3:
```
   |
27 |   installPhase = ''
   |   ^
```
</li>
  </ul>
</details>

Valodim commented:
Cleaned this up again, extracted the patch phase into patches :)

I'm using this for fingerprint authentication on my T460p, like:
```nix
  services.fprintd.enable = true;
  services.fprintd.tod.enable = true;
  services.fprintd.tod.driver = pkgs.libfprint-tod-vfs0090;
  security.pam.services.valodim.fprintAuth = true;
```
